rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -1432,6 +1432,60 @@ private String getTaskResultValueAsStringOrExceptionStackTrace(TaskResult taskRe
     }
 
     /**
+     * Returns the metadata of the task result of task <code>taskName</code> of the
+     * job <code>jobId</code>.
+     *
+     * Metadata is a map containing additional information associated with a result. For example a file name if the result represents a file.
+     *
+     * @param sessionId
+     *            a valid session id
+     * @param jobId
+     *            the id of the job
+     * @param taskname
+     *            the name of the task
+     * @return the metadata of the task result
+     */
+    @GET
+    @GZIP
+    @Path("jobs/{jobid}/tasks/{taskname}/result/metadata")
+    @Produces("*/*")
+    public Map<String, String> metadataOfTaskResult(@HeaderParam("sessionid") String sessionId, @PathParam("jobid") String jobId,
+                                                    @PathParam("taskname") String taskname) throws Throwable {
+        Scheduler s = checkAccess(sessionId, "jobs/" + jobId + "/tasks/" + taskname + "/result/value");
+        TaskResult taskResult = s.getTaskResult(jobId, taskname);
+        taskResult = PAFuture.getFutureValue(taskResult);
+        return taskResult.getMetadata();
+    }
+
+    /**
+     * Returns the metadata of the task result of task <code>taskName</code> of the
+     * job <code>jobId</code>filtered by a given tag.
+     * <p>
+     * Metadata is a map containing additional information associated with a result. For example a file name if the result represents a file.
+     *
+     * @param sessionId a valid session id
+     * @param jobId     the id of the job
+     * @param taskTag   the tag used to filter the tasks.
+     * @return a map containing for each task entry, the metadata of the task result
+     */
+    @GET
+    @GZIP
+    @Path("jobs/{jobid}/tasks/tag/{tasktag}/result/metadata")
+    @Produces("application/json")
+    public Map<String, Map<String, String>> metadataOfTaskResultByTag(@HeaderParam("sessionid") String sessionId,
+                                                                      @PathParam("jobid") String jobId, @PathParam("tasktag") String taskTag) throws Throwable {
+        Scheduler s = checkAccess(sessionId,
+                "jobs/" + jobId + "/tasks/tag" + taskTag + "/result/serializedvalue");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/result/serializedvalue" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1192)